### DISCUSSION TITLE
how can I decode large pkcs7 enveloped data file in a stream manner?

### DISCUSSION THREAD

**Discussion by navigator117** [openssl/openssl#24612] [Q&A]
how can I decode large pkcs7 enveloped data file in a stream manner? like create bio from file, then do symm cipher decode by blocks?

**Comment by navigator117** [comment]
I have tried use mmap backed by pkcs7 file, still got:
140319079558464:error:07069041:memory buffer routines:BUF_MEM_grow_clean:malloc failure:crypto/buffer/buffer.c:131:
140319079558464:error:0D08C041:asn1 encoding routines:collect_data:malloc failure:crypto/asn1/tasn_dec.c:1047:
140319079558464:error:0D08303A:asn1 encoding routines:asn1_template_noexp_d2i:nested asn1 error:crypto/asn1/tasn_dec.c:618:Field=enc_data, Type=PKCS7_ENC_CONTENT
140319079558464:error:0D08303A:asn1 encoding routines:asn1_template_noexp_d2i:nested asn1 error:crypto/asn1/tasn_dec.c:627:Field=enc_data, Type=PKCS7_ENVELOPE
140319079558464:error:0D08303A:asn1 encoding routines:asn1_template_noexp_d2i:nested asn1 error:crypto/asn1/tasn_dec.c:627:
140319079558464:error:0D08403A:asn1 encoding routines:asn1_template_ex_d2i:nested asn1 error:crypto/asn1/tasn_dec.c:477:Field=d.enveloped, Type=PKCS7

**Comment by navigator117** [comment]
It seems openssl just alloc and copy enc_content in d2i_PKCS7, which cause memory alloc failed

**Comment by yhy37092** [comment]
I have encountered the same problem recently. Have you found a solution?
